src/main/java/org/ow2/petals/deployer/utils/RuntimeModelComparator.java

```diff
@@ -0,0 +1,117 @@
+        return compareRuntimeContainerMaps(m1, m2);
+    }
+
+    private static boolean compareRuntimeComponentMaps(RuntimeContainer cont1, RuntimeContainer cont2) {
+        Collection<RuntimeComponent> compList1 = cont1.getComponents();
+        Collection<RuntimeComponent> compList2 = cont2.getComponents();
+
+        for (RuntimeComponent comp1 : compList1) {
+            RuntimeComponent comp2 = cont1.getComponent(comp1.getId());
+            if (comp2 == null || !compareRuntimeComponents(comp1, comp2)) {
+                return false;
+            }
+        }
+
+        for (RuntimeComponent comp2 : compList2) {
+            if (cont2.getComponent(comp2.getId()) == null) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static boolean compareRuntimeContainerMaps(RuntimeModel m1, RuntimeModel m2) {
+        Collection<RuntimeContainer> contList1 = m1.getContainers();
+        Collection<RuntimeContainer> contList2 = m2.getContainers();
+
+        for (RuntimeContainer cont1 : contList1) {
+            RuntimeContainer cont2 = m1.getContainer(cont1.getId());
+            if (cont2 == null || !compareRuntimeContainers(cont1, cont2)) {
+                return false;
+            }
+        }
+
+        for (RuntimeContainer cont2 : contList2) {
+            if (m2.getContainer(cont2.getId()) == null) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static boolean compareRuntimeContainers(RuntimeContainer cont1, RuntimeContainer cont2) {
+        return cont1.getId().equals(cont2.getId()) && cont1.getPort() == cont2.getPort()
+                && cont1.getUser().equals(cont2.getUser()) && cont1.getPassword().equals(cont2.getPassword())
+                && cont1.getHostname().equals(cont2.getHostname()) && compareRuntimeServiceUnitMaps(cont1, cont2)
+                && compareRuntimeComponentMaps(cont1, cont2);
+    }
+
+    private static boolean compareRuntimeServiceUnitMaps(RuntimeContainer cont1, RuntimeContainer cont2) {
+        Collection<RuntimeServiceUnit> suList1 = cont1.getServiceUnits();
+        Collection<RuntimeServiceUnit> suList2 = cont2.getServiceUnits();
+
+        for (RuntimeServiceUnit su1 : suList1) {
+            RuntimeServiceUnit su2 = cont1.getServiceUnit(su1.getId());
+            if (su2 == null || !compareRuntimeServiceUnits(su1, su2)) {
+                return false;
+            }
+        }
+
+        for (RuntimeServiceUnit su2 : suList2) {
+            if (cont2.getServiceUnit(su2.getId()) == null) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static boolean compareRuntimeServiceUnits(RuntimeServiceUnit su1, RuntimeServiceUnit su2) {
+        return su1.getId().equals(su2.getId());
+    }
+
+    private static boolean compareRuntimeComponents(RuntimeComponent comp1, RuntimeComponent comp2) {
+        return comp1.getId().equals(comp2.getId());
```

Comment:
Same here.

Reply:
Why not making an interface that wraps the getId method?
Make RuntimeComponent and RuntimeServiceUnit implement it. You would then have only one comparison method.

Reply:
I don't always want to use just the id